packages/browser/src/ClientAuthentication.ts

```diff
@@ -129,13 +130,39 @@ export default class ClientAuthentication {
     url: string,
     eventEmitter: EventEmitter
   ): Promise<ISessionInfo | undefined> => {
-    const redirectInfo = await this.redirectHandler.handle(url, eventEmitter);
-    // The `FallbackRedirectHandler` directly returns the global `fetch` for
-    // his value, so we should ensure it's bound to `window` rather than to
-    // ClientAuthentication, to avoid the following error:
-    // > 'fetch' called on an object that does not implement interface Window.
-    this.fetch = redirectInfo.fetch.bind(window);
+    try {
+      const redirectInfo = await this.redirectHandler.handle(url, eventEmitter);
+      // The `FallbackRedirectHandler` directly returns the global `fetch` for
+      // his value, so we should ensure it's bound to `window` rather than to
+      // ClientAuthentication, to avoid the following error:
+      // > 'fetch' called on an object that does not implement interface Window.
+      this.fetch = redirectInfo.fetch.bind(window);
+
+      // Strip the oauth params:
+      this.cleanUrlAfterRedirect(url);
+
+      return {
+        isLoggedIn: redirectInfo.isLoggedIn,
+        webId: redirectInfo.webId,
+        sessionId: redirectInfo.sessionId,
+        expirationDate: redirectInfo.expirationDate,
+      };
+    } catch (err) {
+      // Strip the oauth params:
+      this.cleanUrlAfterRedirect(url);
+
+      // FIXME: EVENTS.ERROR should be errorCode, errorDescription
+      //
+      // I'm not sure if "redirect" is a good error code, and in theory `err`
+      // maybe an Error object and not a string; Maybe we want to just hardcode
+      // a description instead?
+      eventEmitter.emit(EVENTS.ERROR, "redirect", err);
+
+      return undefined;
+    }
+  };
 
+  private cleanUrlAfterRedirect(url: string): void {
```

Comment:
This might be better moved to `removeOidcQueryParam` from oidc, but currently that's not how it worked.